diff --git a/browser/components/contextualidentity/content/ContainerCreationPanel.mjs b/browser/components/contextualidentity/content/ContainerCreationPanel.mjs
--- a/browser/components/contextualidentity/content/ContainerCreationPanel.mjs
+++ b/browser/components/contextualidentity/content/ContainerCreationPanel.mjs
@@ -4,7 +4,7 @@
 
 import { ContainerEditor } from "chrome://browser/content/usercontext/ContainerEditor.mjs";
 
-const ANCHOR_ID = "userContext-icons";
+const ANCHOR_ID = "navigator-toolbox";
 const ANCHOR_PINNED_CLASS = "container-anchor-pinned";
 
 const lazy = {};
@@ -69,8 +69,7 @@
     updateValidity();
 
     let onCreate = () => {
-      editor.commit();
-      panel.hidePopup();
+      if (editor.commit() !== false) panel.hidePopup();
     };
     let onCancel = () => panel.hidePopup();
     createButton.addEventListener("click", onCreate);
@@ -95,6 +94,6 @@
       anchor.hidden = false;
     }
 
-    panel.openPopup(anchor, "bottomright topright");
+    panel.openPopup(anchor, win.gZenUIManager.panelUIPosition());
   },
 };
